deploy.py

```diff
@@ -13,105 +13,86 @@ def add_cli_args():
-    cmd = ['ansible-playbook', '-i', 'inventory/inventory.py', 'playbooks/site.yaml']
-
-    if parsed_args.user:
-        cmd += ['-u', parsed_args.user]
+    cmd = []
+
+    if parsed_args.action == 'deploy':
+        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py',
+               'playbooks/site.yaml']
+    elif parsed_args.action == 'new-slave':
+        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py',
+               'playbooks/site.yaml', '-l', '*-slave']
+        parsed_args.extra_vars.append('config_new_slave=true')
+    elif parsed_args.action == 'switch-role':
+        os.environ['OL_SWITCH_MASTER_SLAVE'] = True
+        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py',
+               'playbooks/conf-new-slave.yaml']
+    elif parsed_args.action == 'show-graph':
+        cmd = ['ansible-inventory', '-i', 'inventory/inventory.py', '--graph']
+    elif parsed_args.action == 'show-ip':
+        cmd = ['python', 'inventory/inventory.py', '--show-ip']
 
-    if parsed_args.new_ip:
-        specified_ips = dict([(d.partition('=')[::2]) for d in parsed_args.new_ip])
+    if parsed_args.host_ip:
+        specified_ips = dict([(d.partition('=')[::2])
+                              for d in parsed_args.new_ip])
         for host, ip in specified_ips.items():
             os.environ['OL_%s_IP' % host.upper()] = ip
 
-    if parsed_args.new_slave:
-        if not parsed_args.new_ip:
-            sys.exit("ERROR: Must specify the new slave hosts IPs with --new-ip")
-        os.environ['OL_SWITCH_MASTER_SLAVE'] = True
-        cmd += ['-l', '*-slave']
-        parsed_args.extra_vars.append('config_new_slave=true')
+    if parsed_args.user and cmd[0] == 'ansible-playbook':
+        cmd += ['-u', parsed_args.user]
 
     if parsed_args.extra_vars:
         cmd += ['-e', ' '.join(parsed_args.extra_vars)]
 
-    if parsed_args.switch_role:
-        os.environ['OL_SWITCH_MASTER_SLAVE'] = True
-        cmd = ['ansible-playbook', '-i', 'inventory/inventory.py', 'playbooks/conf-new-slave.yaml']
-
-    if parsed_args.graph:
-        cmd = ['ansible-inventory', '-i', 'inventory/inventory.py', '--graph']
-        if parsed_args.with_vars:
-            cmd.append('--vars')
-
-    if parsed_args.with_vars and not parsed_args.graph:
-        sys.exit("ERROR: --with-vars must work with --graph")
-
-    if parsed_args.show_new_ips:
-        cmd = ['python', 'inventory/inventory.py', '--new-host-ips']
-
-    if parsed_args.show_old_ips:
-        cmd = ['python', 'inventory/inventory.py', '--old-host-ips']
-
-    ol_env_msg = '\n'.join(['%s=%s' % (k, os.environ[k]) for k in os.environ if k.startswith('OL_')])
+    ol_env_msg = '\n'.join(['%s=%s' % (k, os.environ[k]) for k in os.environ
+                            if k.startswith('OL_')])
     print("OpenLab deployment ENV:\n%s" % ol_env_msg)
     print('Ansible command:\n%s' % ' '.join(cmd))
     print("*" * 100)
-
     subprocess.call(cmd)
 
+    if parsed_args.action == 'new-slave':
```

Comment:
how to run config-new-slave.yaml without new slave deployment ?

Reply:
`switch_role` action will run it.

Reply:
OK, got